src/datasets/summarizations/trend-weekly-pattern.summarization.spec.ts

```diff
@@ -0,0 +1,271 @@
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 300 - 5 * i;
+      points.push({ x, y });
+    }
+
+    const isMondayToSundaySummary = isTextPartsInSummary('Monday', 'Sunday', 'similar');
+    const isOtherSummary = norSummaryFilters(
+      isMondayToSundaySummary,
+    );
+
+    const summaries = queryFactory(points)();
+    const mondayToSundaySummary = summaries.filter(isMondayToSundaySummary);
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    expect(mondayToSundaySummary.length).toBe(1);
+    expect(mondayToSundaySummary.every(hasHighValidity)).toBeTrue();
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing constant data.', () => {
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 100;
+      points.push({ x, y });
+    }
+
+    const isMondayToSundaySummary = isTextPartsInSummary('Monday', 'Sunday', 'similar');
+    const isOtherSummary = norSummaryFilters(
+      isMondayToSundaySummary,
+    );
+
+    const summaries = queryFactory(points)();
+    const mondayToSundaySummary = summaries.filter(isMondayToSundaySummary);
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    expect(mondayToSundaySummary.length).toBe(1);
+    expect(mondayToSundaySummary.every(hasHighValidity)).toBeTrue();
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing V shape weekly pattern in constant data.', () => {
+    const weekPattern = [120, 60, 30, 10, 30, 60, 90];  // [Sun, Mon, Tue, Wed, Thu, Fri, Sat]
+    const points: TimeSeriesPoint[] = [];
+    for (let i = 6; i <= 26; i++) {
+      const x = new Date(2020, 6, i);
+      const y = 250 + weekPattern[x.getDay()];
+      points.push({ x, y });
+    }
+
+    const isMondayToWednesdaySummary = isTextPartsInSummary('Monday', 'Wednesday', 'decreased');
+    const isWednesdayToSundaySummary = isTextPartsInSummary('Wednesday', 'Sunday', 'increased');
+    const isOtherSummary = norSummaryFilters(
+      isMondayToWednesdaySummary,
+      isWednesdayToSundaySummary,
+    );
+
+    const summaries = queryFactory(points)();
+    const mondayToWednesdaySummary = summaries.filter(isMondayToWednesdaySummary);
+    const wednesdayToSundaySummary = summaries.filter(isWednesdayToSundaySummary);
+    const otherSummaries = summaries.filter(isOtherSummary);
+
+    expect(mondayToWednesdaySummary.length).toBe(1);
+    expect(mondayToWednesdaySummary.every(hasHighValidity)).toBeTrue();
+    expect(wednesdayToSundaySummary.length).toBe(1);
+    expect(wednesdayToSundaySummary.every(hasHighValidity)).toBeTrue();
+    expect(otherSummaries.every(hasLowValidity)).toBeTrue();
+  });
+
+  it('should create summaries describing V shape weekly pattern in linearly increasing data.', () => {

```

Comment:
"linearly increasing data". Does this mean linearly increase between the same day of the week (i.e. week1 Monday -> week2 Monday -> week3 Monday)

Not easy to understand what the test case is here and for the tests below with "linearly increasing" or "decreasing" data.

Reply:
This is an example of "weekly pattern in linearly increasing data". Basically it is "linear increasing data + pattern offset for the day of the week":
![image](https://user-images.githubusercontent.com/67285909/93140200-e4a88c80-f696-11ea-95bc-04d8f40fbe7d.png)